polaris/curation/viz_utils.py

```diff
@@ -102,3 +140,200 @@ def check_undefined_stereocenters(dataset: PandasDataFrame):
     )
     fig = dm.to_image(to_plot.mol.tolist(), legends=legends)
     return fig
+
+
+def visualize_chemspace(
+    data: pd.DataFrame,
+    mol_col: str = "smiles",
+    size_col=None,
+    seed=1428,
+    umap_metric="jaccard",
+    hue_names: List[str] = None,
+):
+    """
+    Visualize the chemical space by doing a dimensionality reduction with UMAP.
+
+    Args:
+        data: a dataframe
+        split_names: names of the types of splits, should be found in columns of the "data" dataframe
+        mol_col: string column where we will look for a molecule to featurize
+        size_col: used for setting the style of the resulting scatterplot
+        seed: int, random seed to use for reproducibility
+        umap_metric: similarity metric to use for constructing the UMAP representation
+        hue_names: List of grouping variable that will produce points with different colors for the same UMAP plotting.
+
+    Returns:
+        figs, figures for each kind of split specified in split_names
+    """
+    figs = plt.figure(num=3)
+    features = [dm.to_fp(mol) for mol in data[mol_col]]  # Convert molecules to fingerprint
```

Comment:
I would make `features` a parameter, such that people could also use something else than ECFP4. Maybe we can have one parameter `X` and one parameter `smiles` and if `X` is not set, `smiles` needs to be and we will use ECFP4.